lib/adhearsion/call.rb

```diff
@@ -293,6 +301,17 @@ def on_end(&block)
       register_event_handler Punchblock::Event::End, &block
     end
 
+    def terminating!
+      @call_terminating = true
+    end
+
+    #
+    # @return [Boolean] if the call is currently terminating/terminated
+    #
+    def terminating?
```

Comment:
You never actually call this method. Did something go wrong in your refactoring?